src/Core/System/SalesChannel/Validation/SalesChannelValidator.php

```diff
@@ -169,97 +168,68 @@ private function validateLanguages(array $mapping, PreWriteValidationEvent $even
 
         foreach ($mapping as $id => $channel) {
             if (isset($channel['inserts'])) {
-                if (!$this->validInsertCase($channel)) {
+                if (isset($channel['new_default']) && $this->isInvalidInsertCase($channel)) {
                     $inserts[$id] = $channel['new_default'];
                 }
 
                 $duplicatedIds = $this->getDuplicates($channel);
 
-                if ($duplicatedIds) {
+                if ($duplicatedIds !== []) {
                     $duplicates[$id] = $duplicatedIds;
                 }
             }
 
-            if (isset($channel['deletions']) && !$this->validDeleteCase($channel)) {
+            if (isset($channel['deletions'], $channel['current_default']) && $this->isInvalidDeleteCase($channel)) {
                 $deletions[$id] = $channel['current_default'];
```

Comment:
Just wondering if all these special cases are covered by tests. I can only find a integration test for this class and don't know if it runs in all these if blocks 🤔